src/components/settings/Recovery/DelayModifierRow.tsx

```diff
@@ -0,0 +1,62 @@
+import { IconButton, SvgIcon, Tooltip } from '@mui/material'
+import { useContext, useState } from 'react'
+import type { ReactElement } from 'react'
+
+import { TxModalContext } from '@/components/tx-flow'
+import useIsSafeOwner from '@/hooks/useIsSafeOwner'
+import DeleteIcon from '@/public/images/common/delete.svg'
+import EditIcon from '@/public/images/common/edit.svg'
+import CheckWallet from '@/components/common/CheckWallet'
+import { ConfirmRemoveRecoveryModal } from './ConfirmRemoveRecoveryModal'
+import type { RecoveryState } from '@/store/recoverySlice'
+
+export function DelayModifierRow({ delayModifier }: { delayModifier: RecoveryState[number] }): ReactElement | null {
+  const { setTxFlow } = useContext(TxModalContext)
+  const isOwner = useIsSafeOwner()
+  const [confirm, setConfirm] = useState(false)
+
+  if (!isOwner) {
+    return null
+  }
+
+  const onEdit = () => {
+    // TODO: Display flow
+    setTxFlow(undefined)
```

Comment:
I think you forgot to add the UpsertRecoveryFlow here?

Reply:
Indeed, good catch. I've added it in 09cad6f8bc952f8764655756af37d80a1eb86109.